rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/rest/TestExecutionLogService.xtend

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.views.tcltestrun.rest
+
+import java.nio.file.Files
+import javax.json.Json
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.testeditor.rcp4.views.tcltestrun.model.TestExecutionManager
+import javax.json.JsonObjectBuilder
+import java.util.List
+import org.testeditor.rcp4.views.tcltestrun.model.TestLogGroupBuilder
+import org.testeditor.rcp4.views.tcltestrun.model.TestLogGroup
+import org.testeditor.rcp4.views.tcltestrun.model.TestLogGroupComposite
+import javax.json.JsonArray
+
+@Path("/testexeclogs")
+class TestExecutionLogService {
+
+	@Accessors
+	TestExecutionManager testExecutionManager
+
+	@GET
+	@Produces(MediaType::APPLICATION_JSON)
+	def Response getTestLogExeutionsList() {
+		val result = Json.createObjectBuilder
+		val array = Json.createArrayBuilder
+		testExecutionManager.testExecutionLogs.forEach [
+			val execLog = Json.createObjectBuilder
+			execLog.add("filename", it.logFile.name)
+			execLog.add("name", it.testExecutionName)
+			execLog.add("links", createLinks(it.logFile.name))
+			array.add(execLog)
+		]
+		result.add("entries", array)
+		return Response.ok(result.build.toString).build
+	}
+
+	def JsonArray createLinks(String fileName) {
+		val links = Json.createArrayBuilder
+		links.add(Json.createObjectBuilder.add("href", '''/testexeclogs/«fileName»/fulllogs''').add("rel", "fullogs"))
+		links.add(
+			Json.createObjectBuilder.add("href", '''/testexeclogs/«fileName»/logGroups''').add("rel", "logGroups"))
+		links.add(Json.createObjectBuilder.add("href", '''/testexeclogs/«fileName»/logGroups''').add("rel", "self"))
+		return links.build
+	}
+
+	@Path("/{filename}/fulllogs")
+	@GET
+	@Produces(MediaType::APPLICATION_JSON)
+	def Response getTestLogExeutionContent(@PathParam("filename") String filename) {
+		val result = Json.createObjectBuilder
+		val log = testExecutionManager.testExecutionLogs.filter[logFile.name == filename].head
```

Comment:
`TestExecutionManager.getTestExecutionLogs()` will read the entire directory and create an entry for each log until the one that is searched for is found. Shouldn't we pass the `filename` instead to avoid doing some work?
Also the list of `testExecutionLogs` is a good candidate for caching.